apis/kueue/v1alpha1/topology_types.go

```diff
@@ -97,7 +97,7 @@ type TopologySpec struct {
 	// +required
 	// +listType=atomic
 	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:MaxItems=16
```

Comment:
Just out of curiosity - what datacenter layout may require 16 levels of topology?

Reply:
We currently only have use-cases for 8 levels. 

Note that some sub-features of TAS (like node hot swap or taints)  only with `kubernetes.io/hostname` at the lowest level. Also, it is common practice to give some umbrella top-level, like zone. This leaves us with 6 "functional" levels, and in case of modern accelerators, with complex wiring, it might get used.

Since bumping the number of levels requires API change I think it is good to keep some buffer for future unforeseen use-cases, thus proposal of 16.